libqtile/scripts/cmd_obj.py

```diff
@@ -40,6 +40,23 @@
 from libqtile.ipc import Client, find_sockfile
 
 
+class KeyValDictAdd(argparse.Action):
+    def __call__(self, parser, args, values, option_string=None):  # noqa: F841
+        kwargs = {}
+        for value in values:
+            k, _, v = value.partition("=")
+            kwargs[k] = v
+
+        setattr(args, self.dest, kwargs)
+
+
+def keyval(value):
+    k, _, v = value.partition("=")
+    if not k and v:
+        raise argparse.ArgumentTypeError(f"{value} is not a valid key=value pair.")
+    return value
```

Comment:
Actually, this is probably redundant as we can deal with this during the custom action parsing.